### PR TITLE
🪲 Use `getNextNonce` when proposing transactions to safe

### PR DESCRIPTION
**Pull request by janjakubnanista**
### In this PR

- Add `nonce` coming from `getNextNonce` when proposing transactions to gnosis

### CODE REVIEW

**Review by lzBryan — APPROVED** [review]

File: packages/devtools-evm/src/signer/sdk.ts [hunk 1]
```diff
@@ -130,23 +134,20 @@ export class GnosisOmniSignerEVM<TSafeConfig extends ConnectSafeConfig> extends
...
                 ethAdapter,
                 safeAddress: this.safeConfig.safeAddress!,
-                ...(!!contractNetworks && { contractNetworks }),
+                contractNetworks: this.safeConfig.contractNetworks,
             })
         }
-        if (!this.safeSdk || !this.apiKit) {
-            throw new Error('Safe SDK not initialized')
-        }
+
```
Comment (TRileySchwarz) [inline packages/devtools-evm/src/signer/sdk.ts:150]: log msg for already init?

Comment (TRileySchwarz) [reply packages/devtools-evm/src/signer/sdk.ts:150]: else case~

Comment (TRileySchwarz) [reply packages/devtools-evm/src/signer/sdk.ts:150]: might clog the debug if theres hundreds of tx though, so maybe not

**Comment by TRileySchwarz** [comment]
This only addresses the nonce concept, not the multisender address for now correct?

File: packages/devtools-evm/src/signer/sdk.ts (hunk 1 above)
Comment (janjakubnanista) [reply packages/devtools-evm/src/signer/sdk.ts:150]: that's gonna log on every transaction, this method is being called for every `signAndSend`

**Comment by janjakubnanista** [comment]
> This only addresses the nonce concept, not the multisender address for now correct?

Oh that's already supported - we just need to set `contractNetworks` in `safeConfig` in hardhat config

File: packages/devtools-evm/src/signer/sdk.ts (hunk 1 above)
Comment (TRileySchwarz) [reply packages/devtools-evm/src/signer/sdk.ts:150]: yeah realized that, nvm

**Review by TRileySchwarz — APPROVED** [review]

**Pull request merged**